### PR TITLE
Missing parts of the API for repository issues

### PR DESCRIPTION
**Pull request by andreasabel**
Motivation: see issue #286 and https://github.com/agda/agda/pull/5691
Closes: #286 and #287
Needs: release (e.g. 0.28)

As observed in #286 and #287, the API for listing repository issues, see https://docs.github.com/en/rest/reference/issues#list-repository-issues, is incompletely reflected in `github-0.27`.  Esp. the functionality to get just the issues for a _particular_ milestone is not exported.

This PR adds the missing parts of the API as new methods to construct `IssueRepoMod`.
With the functionality added by this PR, the API is now complete.  In particular, I conjecture the following completeness statement:  
> Using the language exported by `GitHub.Data.Options`,
> for all _o_ : `IssueRepoOptions`, there is some _m_ : `IssueRepoMod`, such that _o_ = `toIssueRepoOptions` _m_.

In contrast, the API of 0.27 is incomplete: 
> There is are no _m_ : `IssueRepoMod` and _i_ : `Id Milestone` such that `toIssueRepoOptions` _m_ = `defaultIssueRepoOptions{ issueRepoOptionsMilestone = FilterBy` _i_ `}`.

Successful CI run at: https://github.com/andreasabel/github/actions/runs/1556147789

### CODE REVIEW

**Comment by andreasabel** [comment]
Ping @phadej.  The workflow needs to be started manually.

File: src/GitHub/Data/Options.hs [hunk 1]
```diff
@@ -491,16 +495,70 @@ issueFilter f = IssueMod $ \opts ->
 -- Issues repo
 -------------------------------------------------------------------------------
 
+-- | See <https://docs.github.com/en/rest/reference/issues#list-repository-issues>.
+--   Retrieved: 2021-12-08
+--
+-- Parameters of "list repository issues" (@get /repos/{owner}/{repo}/issues@)
```
Comment (phadej) [inline src/GitHub/Data/Options.hs:501]: Oh please don't copy the GitHub docs. That's why there is a link. It's a maintenance burden without an easy way to check if something had changed.

File: github.cabal [hunk 2]
```diff
@@ -1,7 +1,6 @@
 cabal-version:      >=1.10
 name:               github
-version:            0.27
-x-revision:         1
+version:            0.28
```
Comment (phadej) [inline github.cabal:3]: Where is a breaking  change?

Comment (andreasabel) [reply github.cabal]: No breaking change, only API addition.  So 0.27.1 should be fine.

**Comment by andreasabel** [comment]
Thanks for the review, @phadej!  I updated the PR accordingly.

**Comment by andreasabel** [comment]
@phadej, what's the plan?  Can you release this fix?

**Comment by phadej** [comment]
@andreasabel do you want to be come the maintainer of `github`. I'll happily transfer this repository to you.

**Comment by andreasabel** [comment]
> @andreasabel do you want to be come the maintainer of `github`. I'll be happy to transfer this repository to you.
> 
> (I have been looking for someone to look after this library for quite some time)

@phadej Yes, I am happy to maintain `github`.  I have some interest in this library.

**Comment by andreasabel** [comment]
@phadej ping.

**Comment by phadej** [comment]
I cannot transfer this repository as *andreasabel/github already exists*

**Comment by phadej** [comment]
Looks like that's not enough, it still errors with 

> andreasabel already has a repository in the phadej/github network

**Pull request merged**